api-extractor/src/definitions/ApiNamespace.ts

```diff
@@ -0,0 +1,108 @@
+/* tslint:disable:no-bitwise */
+
+import * as ts from 'typescript';
+import ApiField from './ApiField';
+import ApiItem, { ApiItemKind, IApiItemOptions } from './ApiItem';
+import ApiItemContainer from './ApiItemContainer';
+import { IExportedSymbol } from '../IExportedSymbol';
+
+const allowedTypes: string[] = ['string', 'number', 'boolean'];
+
+/**
+  * This class is part of the ApiItem abstract syntax tree. It represents exports of
+  * a namespace, the exports can be classes, interfaces, type literals, enums, functions,
+  * and properties.
+  */
+export default class ApiNamespace extends ApiItemContainer {
+  private _exportedNormalizedSymbols: IExportedSymbol[] = [];
+
+  constructor(options: IApiItemOptions) {
+    super(options);
+    this.kind = ApiItemKind.Namespace;
+    this.name = options.declarationSymbol.name;
+
+    const exportSymbols: ts.Symbol[] = this.typeChecker.getExportsOfModule(this.declarationSymbol);
+    if (exportSymbols) {
+      for (const exportSymbol of exportSymbols) {
+        const followedSymbol: ts.Symbol = this.followAliases(exportSymbol);
+
+        if (!followedSymbol.declarations) {
+          this.reportWarning(`Definition with no declarations: ${exportSymbol.name}`);
+          continue;
+        }
+
+        if (!(followedSymbol.flags === ts.SymbolFlags.BlockScopedVariable)) {
+          this.reportWarning(`Unsupported export "${exportSymbol.name}" ` +
+            'ApiNamespace only supports properties.');
```

Comment:
"Unsupported export ____. Currently the "namespace" block only supports constant variables."

---
In reply to: [114371996](https://github.com/Microsoft/web-build-tools/pull/167#discussion_r114371996) [](ancestors = 114371996)